### ISSUE TITLE
"Couldn't access GitHub API"

### ISSUE THREAD

**Issue by Studio384** [yosko/php-github-updater#1]
Hey,

I'm trying to implement this project into mine as an update system, but it doesn't work. When I use the advanced example, I get the message that it couldn't access the GitHub API. Is this a problem on my side or has the GitHub API changed in the meantime? If the first, could you please help me fixing it? If the later, are you able to create an update to make this work again? I do not have much experience with the GitHub API.

Thanks in advance!

**Comment by yosko** [comment]
Hey,

Which Github repository are you trying to check? Are you sure there are releases on this project (releases are not the same thing as tags).

Are you behind a proxy or something? In that case, you should make a call to `useProxy` prior to any other method call.

I haven't worked on this project for more than a year, so it wouldn't be surprising if the Github API had changed.
But so far, all my tests (using the repository yosko/jotter) seem to work including php-github-updater's following methods:
- `installLatestVersion`
- `downloadVersion`
- `extractArchive`
- `moveFilesRecursive`
- `getLatestVersion`
- `getTitle`
- `getDescription`
- `getReleases`
- `isUpToDate`
- `fetchPrereleasesToo`

The method `getNextVersion` seems to give latest version instead of the next one, so even if it is wrong, it still manages to contact github too.

As for the other methods, they are called within the ones I mentionned before, so they seem to work too (`compareVersions`, `getZipballUrl`, `getTarballUrl`, `getContentFromGithub`).

**Comment by Studio384** [comment]
I'm trying to get updates from GetLuna/LunaUpdateDrive as a test for GetLuna/Luna, and there are releases present, I'll try the proxy-fix first. Thanks for responding and thumbs up for this project.

**Comment by yosko** [comment]
It might be different from the problem you have, but I ran my previous tests with your repository and got some error at first (using `$localVersion = 'Build 4681';`).

But then I realised that this was the "title" of the release, not its technical name. When trying with `$localVersion = '1.1.4681';`, everything works just fine.

php-github-updater needs some more controls for input data (repository and release names).

**Comment by Studio384** [comment]
I myself use the '1.1.4681' naming too. Are you able to get 1.1.4682 from GitHub (the second tag?).

Anyway, any idea on how to determine the IP and port?

**Comment by yosko** [comment]
Yes, I am able to download it. Here is the working snippet I used:

``` php
$updater = new PhpGithubUpdater('GetLuna', 'LunaUpdateDrive');
$archive = $updater->downloadVersion('1.1.4682', 'temp-directory');
```

As for the proxy: it is not about Github's proxy, but yours. If your server uses a proxy, you might need to use `useProxy` but I can't guess the values for you.

Finally, if you are using this code in localhost, you might be unable to access external resources (such as Github's) due to your web server and PHP configuration or even your system's, and I can't be of any help on that. You might wan't to try from a "real" server to see the difference.

What the exact error message you get? Is there a line number given?

**Comment by Studio384** [comment]
I know, what I actualy asked was wheter you know a way to know the IP and port of someone that installs this on their server, can't simply hardcode mine into the code ofcourse.

Anyway, no, there is no error given on a live server. The script doesn't return anything. I'm going to try a couple of other tests and report back here if it works or not (and if it might return an error).

**Comment by ninjagolloyd** [comment]
It is a year old. Has github been updated..?

**Comment by yosko** [comment]
@Studio384 : I'm sorry, no, I don't know how to get that information.

@ninjagolloyd : as mentioned in my first post, apparently no. I retested every method of the library, and everything still seems to work.

**Comment by Studio384** [comment]
Well then, if there is no way to get to that, I asume the road for this feature has come to an end here. For now that is... This has been draging on Luna 1.1 for too long already...

Anyway, how do systems like WordPress handle this...
